src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseReport.java

```diff
@@ -0,0 +1,603 @@
+
+    row = maybeInList("Tissue", labelListSlides, thisMonthSlides, rowNum);
+    if (row.length != 0) return row;
+    rowNum -= (thisMonthSlides.size() + 4); // list + (blank, 2 headers, total)
+
+    if (rowNum == 0) {
+      return makeBlankRow();
+    } else if (rowNum == 1) {
+      return makeAlmostBlankRow(String.format("Total Inventory Available (%s)", new SimpleDateFormat(DATE_FORMAT).format(new Date())));
+    }
+    rowNum -= 2;
+
+    row = maybeInList("Total in Inventory", labelListInventory, inventoryAvailable, rowNum);
+    if (row.length != 0) return row;
+    rowNum -= (inventoryAvailable.size() + 4); // list + (blank, 2 headers, total)
+
+    if (rowNum == 0) {
+      return makeBlankRow();
+    } else if (rowNum == 1) {
+      String[] blank = makeBlankRow();
+      blank[0] = "Samples with mismatched alias and cryovial/slide number";
+      return blank;
+    }
+
+    row = maybeSampleAlias(barcodeAndNameCodingMismatch, rowNum);
+    if (row.length != 0) return row;
+    rowNum -= barcodeAndNameCodingMismatch.size();
+
+    if (rowNum == 0) {
+      return makeBlankRow();
+    } else if (rowNum == 1) {
+      String[] blank = makeBlankRow();
+      blank[0] = "Slides with non-slide timepoint in alias";
+      return blank;
+    }
+
+    row = maybeSampleAlias(slidesWithoutSlideTimepoint, rowNum);
+    if (row.length != 0) return row;
+    rowNum -= slidesWithoutSlideTimepoint.size();
+
+    return makeBlankRow();
+  }
+
+  private String[] maybeInList(String category, List<SampleLabel> labels, List<List<String>> list, int rowNum) {
+    if (rowNum == 0) {
+      return makeBlankRow();
+    } else if (rowNum == 1) {
+      return generateRow(labelListHeader(category, labels));
+    } else if (rowNum == 2) {
+      return generateRow(sampleCasesHeader(labels));
+    }
+    rowNum -= 3;
+    if (rowNum < list.size()) {
+      return generateRow(list.get(rowNum));
+    } else if (rowNum == list.size()) {
+      return generateTotalsRow(list, labels);
+    }
+    return new String[0];
+    // do our subtraction of the list + totals outside this method, because changes here won't affect rowNum outside the method
+  }
+
+  private String[] maybeSampleAlias(List<SampleDto> samples, int rowNum) {
+    if (rowNum < samples.size()) {
+      SampleDto sample = samples.get(rowNum);
+      String[] row = makeAlmostBlankRow(sample.getName());
+      row[1] = sample.getTubeBarcode();
+      return row;
+    }
+    return new String[0];
+  }
+
+  private String[] generateRow(List<String> received) {
+    String[] row = makeBlankRow();
+    for (int i = 0; i < received.size(); i++) {
+      row[i] = received.get(i);
+    }
+    return row;
+  }
+
+  private String[] generateZerosRow(String firstColumn, List<SampleLabel> labels) {
+    String[] zeros = makeAlmostBlankRow(firstColumn);
+    for (int i = 1; i < zeros.length; i++) {
+      zeros[i] = i <= 2 * labels.size() ? "0" : "";
```

Comment:
**`makeAlmostBlankRow` already fills the array with empty Strings. Could just change to
```
for (int i = 1; i > 2*labels.size(); i++) {
  zeros[i] = "0";
}
```